### DISCUSSION TITLE
SAML with google

### DISCUSSION THREAD

**Discussion by bhagya1101** [directus/directus#16887] [Archive]
How to perform SAML with google in directus? Is there any reference to it or any demo/example repository that performs SAML using directus??
Its getting somewhat complicated for me .
Any help will be appreciated.

**Comment by alexvdvalk** [comment]
Can you describe your current setup? Self hosted / Directus Cloud?

Our SSO documentation can be found here: https://docs.directus.io/self-hosted/sso-examples.html

**Reply by bhagya1101** [reply]
Its self hosted and also if I want to retrieve access_token of current_user in directus custom module , then how should I do it because I have created card and grid display and on click of that card I want that clicked data to be added on my collection

**Reply by rijkvanzanten** [reply]
You can use the `useApi()` composable exported form the extensions SDK to use the current authenticated user to query the API:

```js
import { useApi } from '@directus/extensions-sdk';

const api = useApi();

await api.post('/items/cards', { name: 'my new card' });
```

**Reply by bhagya1101** [reply]
I did ' npm init directus-extension' and created a module. In the module.vue file of the generated module, I added the above code of useApi. It is giving me not authenticated. I dont want to give my collection public role.

I used 
await api.post('/items/resources', { name : 'my new card' } ) and it gave me unauthorized error.

**Reply by bhagya1101** [reply]
Any help will be appreciated.
Thanks.

**Reply by bhagya1101** [reply]
Thanks a lot. It is now done. 
Really thanks a lot..
